dimod/include/dimod/quadratic_model.h

```diff
@@ -33,6 +33,89 @@ enum Vartype {
     INTEGER  ///< Variables that are integer valued.
 };
 
+/// Compile-time limits by variable type.
+template <class Bias, Vartype vartype>
+class vartype_limits {};
+
+template <class Bias>
+class vartype_limits<Bias, Vartype::BINARY> {
+ public:
+    static constexpr Bias default_max() noexcept { return 1; }
```

Comment:
Unlike `Integer` variables, having a `default_min` and `default_max` seems a little off for both `Binary` and `Spin` variables. It seems to imply that there are non-defaults options that are allowed.

Reply:
Agree it's a bit verbose, but it lets us write more generic code. For instance https://github.com/dwavesystems/dimod/pull/1094/files#diff-314bf3065935756ec61ae283725d32f27810d08fb64dd440c4195b17fc2e390eR1076 lower down.
```c++
        return this->add_variable(
                vartype, vartype_info<bias_type>::default_min(vartype),
                vartype_info<bias_type>::default_max(vartype));
```

Reply:
This is overall following the pattern of [numeric_limits](https://en.cppreference.com/w/cpp/types/numeric_limits), which for instance also supports `bool` even though most of the values are just 0/1